carma-messenger-core/carma-messenger/launch/carma-messenger.launch.py

```diff
@@ -0,0 +1,109 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ament_index_python import get_package_share_directory
+from launch.actions import Shutdown
+from launch import LaunchDescription
+from launch_ros.actions import Node
+from launch.actions import IncludeLaunchDescription
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.substitutions import ThisLaunchFileDir
+from launch.substitutions import EnvironmentVariable
+from launch.actions import GroupAction
+from launch_ros.actions import PushRosNamespace
+from carma_ros2_utils.launch.get_log_level import GetLogLevel
+from launch.substitutions import LaunchConfiguration
+from launch.actions import DeclareLaunchArgument
+
+import os
+
+def generate_launch_description():
+    """
+    Launch CARMA System.
+    """
+
+    system_controller_param_file = os.path.join(
+        get_package_share_directory('system_controller'), 'config/config.yaml')
+
+    env_log_levels = EnvironmentVariable('CARMA_ROS_LOGGING_CONFIG', default_value='{ "default_level" : "WARN" }')
+
+    vehicle_calibration_dir = LaunchConfiguration('vehicle_calibration_dir')
+    declare_vehicle_calibration_dir_arg = DeclareLaunchArgument(
+        name = 'vehicle_calibration_dir', 
+        default_value = "/opt/carma/vehicle/calibration",
+        description = "Path to folder containing vehicle calibration directories"
+    )
+
+    vehicle_config_dir = LaunchConfiguration('vehicle_config_dir')
+    declare_vehicle_config_dir_arg = DeclareLaunchArgument(
+        name = 'vehicle_config_dir', 
+        default_value = "/opt/carma/vehicle/config",
+        description = "Path to file containing vehicle config directories"
+    )
+
+    # Declare the vehicle_calibration_dir launch argument
+    vehicle_characteristics_param_file = LaunchConfiguration('vehicle_characteristics_param_file')
+    declare_vehicle_characteristics_param_file_arg = DeclareLaunchArgument(
+        name = 'vehicle_characteristics_param_file', 
+        default_value = [vehicle_calibration_dir, "/identifiers/UniqueVehicleParams.yaml"],
+        description = "Path to file containing unique vehicle characteristics"
+    )
+
+    # Declare the vehicle_config_param_file launch argument
+    vehicle_config_param_file = LaunchConfiguration('vehicle_config_param_file')
+    declare_vehicle_config_param_file_arg = DeclareLaunchArgument(
+        name = 'vehicle_config_param_file',
+        default_value = [vehicle_config_dir, "/VehicleConfigParams.yaml"],
+        description = "Path to file contain vehicle configuration parameters"
+    )
+
+    v2x_group = GroupAction(
+        actions=[
+            PushRosNamespace(EnvironmentVariable('CARMA_MSG_NS', default_value='message')),
+            IncludeLaunchDescription(
+                PythonLaunchDescriptionSource([ThisLaunchFileDir(), '/message.launch.py']),
```

Comment:
It doesn't look like there is a `message.launch.py` file located in this file's directory. There is a chance I am misunderstanding the usage of `ThisLaunchFileDir()`, but I still wanted to point this out.

Reply:
Big oversight on my part. I had neglected to actually launch the nodes. I created this file and implemented a simple set of triggers for the lifecycle. At the moment, I'm not setting up a system_controller for carma-messenger since it didn't even have a health_monitor before and therefore its out of scope to add it now given the time constraints we have. Hopefully we can get to it in the future.